mlflow/tracking/request_auth/abstract_request_auth_provider.py

```diff
@@ -0,0 +1,34 @@
+from abc import ABCMeta, abstractmethod
+
+from mlflow.utils.annotations import developer_stable
+
+
+@developer_stable
+class RequestAuthProvider(ABC):
+    """
+    Abstract base class for specifying custom request auth to add to outgoing requests
+
+    When a request is sent, MLflow will iterate through all registered RequestAuthProviders.
+    For each provider where ``get_name`` matches auth provider name, MLflow calls the ``get_auth``
+    method on the provider to compute request auth.
+
+    The resulting request auth will then be added and sent with the request.
+    """
+
+    @abstractmethod
+    def get_name(self):
+        """
+        Get the name of the request auth provider.
+
+        :return: str of request auth provider name
+        """
+        pass
+
+    @abstractmethod
+    def get_auth(self):
+        """
+        Generate request auth object.
```

Comment:
```suggestion
        Generate request auth object (e.g., ...)
```

Can we add an example value?

Reply:
As this value is up to vendors, can we add a link to the `authentication` of python's `request` library? https://requests.readthedocs.io/en/latest/user/authentication/

Reply:
can we add type hints if possible?

Reply:
> As this value is up to vendors, can we add a link to the authentication of python's request library?

Sounds good

Reply:
> can we add type hints if possible?

As the `get_auth` could be an object/tuple/dict, it seems not likely to do the typing 😅 , I can add the typing for `get_name`

Reply:
Got it, then just add an example. `request auth object` sounds ambiguous.

Reply:
```suggestion
        Generate request auth object (e.g., `requests.auth import HTTPBasicAuth`). See
        https://requests.readthedocs.io/en/latest/user/authentication/ for more details.
```

Reply:
Thank you I was trying to commit exactly the same change 😄